File: DiffProg_MyBook/julia_lab/Basic/test_struct_r1/test_struct.rs
struct Point {
    x: i32,
    y: i32
}

impl std::fmt::Display for Point {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        write!(f, "Point({}, {})", self.x, self.y)
    }
}

fn test1() {
    let p1 = Point {x: 1, y: 2};
    println!("{}", p1);
}

fn main() {
    test1();
}

File: DiffProg_MyBook/julia_lab/Basic/test_struct_r1/test_struct.cpp
#include <iostream>

struct Point {
    int x;
    int y;
};

void test1() {
    struct Point p1; //= Point(1, 2);
    p1.x = 1;
    p1.y = 2;
    
    std::cout << "(" << p1.x << "," << p1.y << ")" << std::endl;
}

int main() {
    test1();
    return 0;
}

